python/ray/tune/trainable.py

```diff
@@ -405,6 +418,53 @@ def _restore(self, checkpoint):
 
         raise NotImplementedError
 
+    def _delete(self, checkpoint_dir):
+        """Removes directory and checkpoint from disk.
+
+        Parameters
+        ----------
+            checkpoint_dir (str): The directory where the checkpoint
+                file is placed.
+
+        """
+        shutil.rmtree(checkpoint_dir)
+
+    def _checkpoint_high_reward(self, checkpoint_dir):
```

Comment:
Move everything except delete_checkpoint to tune, probably somewhere in `ray_trial_executor.py`